x/market/handler/endblock.go

```diff
@@ -78,19 +125,16 @@ func matchOrders(ctx sdk.Context, keepers Keepers) error {
 			return false
 		}
 
-		// open bids; match by lowest price
-		sort.Slice(bids, func(i, j int) bool {
-			// TODO handle same price
-			return bids[i].Price.IsLT(bids[j].Price)
-		})
-
-		winner := bids[0]
+		winner, err := pickBidWinner(bids)
+		if err != nil {
+			return false
```

Comment:
should probably panic here :cry:

Reply:
So in this case, returning false would block iterating over all the orders if I understand correctly?
https://github.com/ovrclk/akash/blob/2b84c28a07d70653bc35543cf27d90eae0378597/x/market/keeper/keeper.go#L295-L301

Normally I always balk at using panics, but I'm not sure the SDK gives us a better option sadly... :disappointed:

Reply:
If there's an error when trying to match orders then the state is broken - in cosmos you deal with that by blowing up.